CDCgov/prime-reportstream
# Cfelix/2485 cli lookup tables

This PR ...

**If you are suggesting a fix for a currently exploitable issue, please disclose the issue to the prime-reportstream team directly outside of GitHub instead of filing a PR, so we may immediately patch the affected systems before a disclosure. See [SECURITY.md/Reporting a Vulnerability](https://github.com/CDCgov/prime-reportstream/blob/master/SECURITY.md#reporting-a-vulnerability) for more information.**

Test Steps:
1. Build the branch by running `gradle package`
2. Run `./prime lookuptable --help` and verify the help text with the available commands is displayed.
3. Run `./prime lookuptable --c list` to get the list of tables and verify no tables as found as the database is currently empty.
4. Load a new table by running the command `./prime --c create --name fips-county --i ./metadata/tables/fips-county.csv`.  Verify that the data is shown in table format.  Answer the prompt with 'Y' and verify the table was created.
5. Run `./prime lookuptable --c list --a` and verify the fips-county table version 1 is present and inactive.
6. Activate the table by running `./prime lookuptable --c activate --name fips-county --version 1` and answer 'Y' on the prompt. 
7. Run `./prime lookuptable --c list` and verify the fips-county table version 1 is present and active.
8. Copy the ./metadata/tables/fips-county.csv file to a temporary location and delete a few lines and change others.  This will make a revised table for our test.
9. Create a new version of the table by running `./prime --c create --name fips-county --i <your temp file>`.  Verify that the data is shown in table format.  Answer the prompt with 'Y' and verify the table was created.
10. Run `./prime lookuptable --c list --a` and verify that both table versions are shown with version 1 still active.
11. Activate the table by running `./prime lookuptable --c activate --name fips-county --version 2` and answer 'Y' on the prompt.
12. Run `./prime lookuptable --c list --a` and verify that both table versions are shown with version 2 now active.
13. Compare table versions by running `./prime lookuptable --c diff --name fips-county --v1 1 --v2 2` and verify the chages you made are shown as a diff.
14. Deactivate the table by running `./prime lookuptable --c deactivate --name fips-county` and answer 'Y' on the prompt.
15. Run `./prime lookuptable --c list --a` and verify that both table versions are shown as inactive.
16. Run './prime lookuptable --c get --name fips-county` and verify version 2 of the table is displayed as a table on the screen.
17. Run './prime lookuptable --c get --name fips-county --o ./fips-output-test.csv` and verify a file is written fips-output-test.csv in CSV format with the table data for version 2 of the table.

## Changes
- *Include a comprehensive list of changes in this PR*
-

## Checklist

### Testing

### Process

## Fixes
- #issue - List GitHub issues this PR fixes

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
